include/mbedtls/ssl.h

```diff
@@ -204,6 +204,9 @@
 #define MBEDTLS_SSL_DTLS_SRTP_MKI_UNSUPPORTED    0
 #define MBEDTLS_SSL_DTLS_SRTP_MKI_SUPPORTED      1
 
+#define MBEDTLS_SSL_SRV_RESPECT_CLIENT_PREF_ENABLED    1
+#define MBEDTLS_SSL_SRV_RESPECT_CLIENT_PREF_DISABLED   0
```

Comment:
Naming: this is both vague (what preferences?) and complex (what does “enabled” vs “disabled” mean?). How about
```
#define MBEDTLS_SSL_SRV_CIPHERSUITE_ORDER_CLIENT 1
#define MBEDTLS_SSL_SRV_CIPHERSUITE_ORDER_SERVER 0
```

Note: my proposal is not a good set of names if we think these constants may be used for preferences other than the ciphersuite order in the future. @hanno-arm ?